### PR TITLE
Fix false negatives for ignore: ["inside-block"] in rule-empty-line-before

### PR DESCRIPTION
**Pull request by gucong3000**
> Which issue, if any, is this issue related to?

#2891

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/__tests__/fixtures/rule-empty-line-before.html [hunk 1]
```diff
@@ -5,6 +5,9 @@
 h1 {
```
Comment (jeddy3) [inline lib/__tests__/fixtures/rule-empty-line-before.html]: Would the following?

```css
@media (min-width: 100px) {
  h1 {
    font-weight: normal;
  }
}
```

And no warnings be a more appropriate test for this change as your fixing false positives, right?

Comment (gucong3000) [reply lib/__tests__/fixtures/rule-empty-line-before.html]: > Would the following?
It has nothing to do with this.
The affected is that the <style> tag first-level child element is misjudged as nested.

> And no warnings be a more appropriate test for this change as your fixing false positives, right?

It is not "false positives", It is "missed warnings"
(I don't know if my English is appropriate, title modified, please help correct English grammar)

Comment (jeddy3) [reply lib/__tests__/fixtures/rule-empty-line-before.html]: > The affected is that the <style> tag first-level child element is misjudged as nested.

Oooo, I see now. Sorry, I misunderstood. 

> It is not "false positives", It is "missed warnings"

I've just been using "false negatives" for those :)

File: lib/__tests__/standalone-syntax.test.js [hunk 2]
```diff
@@ -120,8 +125,12 @@ it("standalone with postcss-html syntax", () => {
     const ruleEmptyLineBeforeResult = results.find(r =>
       /[/\\]rule-empty-line-before\.html$/.test(r.source)
     );
-    expect(ruleEmptyLineBeforeResult.errored).toBeFalsy();
-    expect(ruleEmptyLineBeforeResult.warnings.length).toBe(0);
+    expect(ruleEmptyLineBeforeResult.errored).toBe(true);
+    expect(ruleEmptyLineBeforeResult.warnings.length).toBe(1);
+    expect(ruleEmptyLineBeforeResult.warnings[0].line).toBe(8);
```
Comment (ntwb) [inline lib/__tests__/standalone-syntax.test.js]: What are the 8 warnings added in this change?

Comment (gucong3000) [reply lib/__tests__/standalone-syntax.test.js]: not 8 warnings, [just 1](https://github.com/stylelint/stylelint/pull/2894/files/51e7a7f7b8dfa93e0bc1dc4da241768a06eaef4f#diff-7d31f5d52b54f165816fa0ff4e0de2dfR132).

**Comment by jeddy3** [comment]
> -   Fixed: `rule-empty-line-before` false positives for `ignore: ["inside-block"]` and CSS in HTML ([#2894](https://github.com/stylelint/stylelint/issues/2894)).

**Pull request merged**